### PR TITLE
feat(packaging): add subpackage scaffolding, link MIGRATION, fix audit recursion

### LINKED ISSUE #1554: [audit][nitpick] Packaging/DX gaps: MIGRATION link, OS CI, recursive audit task, no scaffolding
- **[S12 Packaging]** MIGRATION.md not linked from README.md — `README.md`
- **[S12 Packaging]** macOS/Windows CI coverage is documented-absent (tracked issue #539) — `.github/workflows/test.yml:7-12`
- **[S13 Developer Experience]** `pixi.toml` [tasks] `audit` task uses `pixi run --environment lint` inside a pixi task (recursive invocation) — `pixi.toml:18`
- **[S13 Developer Experience]** No codegen/scaffolding tooling for new subpackages or new CLI scripts — `scripts/`

Split from #1518 (NITPICK audit bundle, strict full-coverage audit 2026-06-16). Low priority; address opportunistically.

### PR DESCRIPTION
**Pull request by mvillmow**
## Summary
Address four packaging and DX gaps from the #1518 audit: link MIGRATION.md in README, document macOS/Windows CI coverage gap, remove recursive pixi invocation in audit task, and add subpackage/script scaffolding tooling.

## Changes
- Link MIGRATION.md from README.md and docs/index.md for discoverability
- Document macOS/Windows CI coverage as tracked by issue #539 in COMPATIBILITY.md
- Replace recursive `pixi run --environment lint` in audit task with direct `ruff check` invocation
- Add `scaffold_subpackage.py` library and CLI script for generating new hephaestus subpackages with standard structure
- Register scaffold_subpackage in pyproject.toml console scripts and pixi tasks

## Testing
- Unit tests for scaffold_subpackage library (parameter validation, file generation, edge cases)
- Manual verification: `pixi run scaffold-subpackage test_new_pkg` creates valid package structure
- Verify audit task runs without recursive pixi invocation: `pixi run audit`
- Confirm MIGRATION.md link resolves in rendered README and docs

## Closes
Closes #1554

Generated by Claude Code via ProjectHephaestus automation.

### CODE REVIEW

File: hephaestus/scripts_lib/scaffold_subpackage.py [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+
+    Returns:
+        Exit code: 0 on success, 1 on error.
+
+    """
+    parser = argparse.ArgumentParser(
+        prog="hephaestus-scaffold-subpackage",
+        description="Scaffold a new hephaestus subpackage skeleton with matching test directory.",
+    )
+    parser.add_argument(
```
Comment (mvillmow) [inline hephaestus/scripts_lib/scaffold_subpackage.py:134]: This new console script registers in pyproject.toml [project.scripts] but only calls add_json_arg(parser) — it never calls add_version_arg(parser). tests/integration/test_cli_entry_points.py::TestCLIVersionFlag auto-parametrizes over EVERY entry point and asserts both `--version` and `-V` exit 0 with the command name in output (lines 136-173). argparse will reject `--version`/`-V` here with exit code 2, so test_version_flag and test_version_flag_short_form will FAIL for hephaestus-scaffold-subpackage. Fix: import add_version_arg from hephaestus.cli.utils and call add_version_arg(parser) alongside add_json_arg(parser) in main().

**Review by mvillmow — COMMENTED** [review]
NOGO: new console script omits add_version_arg; TestCLIVersionFlag requires --version/-V on every [project.scripts] entry, so two integration tests will fail. Items 1-3 are correct.

File: tests/unit/scripts_lib/test_scaffold_subpackage.py [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+    def test_invalid_name_nonzero_exit(self, name: str, tmp_path: Path) -> None:
+        args = [name] if name else []
+        rc = main(["--root", str(tmp_path), *args]) if name else 1
+        assert rc != 0
+
+    @pytest.mark.parametrize("name", ["Bad-Name", "1abc", "CamelCase"])
+    def test_invalid_name_writes_nothing(self, name: str, tmp_path: Path) -> None:
+        main(["--root", str(tmp_path), name])
+        assert not (tmp_path / "hephaestus").exists()
+
```
Comment (mvillmow) [inline tests/unit/scripts_lib/test_scaffold_subpackage.py:79]: The empty-name parametrize case (`""`) never exercises `main()`: the ternary `rc = main(...) if name else 1` short-circuits to a hardcoded `1` for the empty string, so this case asserts `1 != 0` tautologically and gives zero coverage of the empty-name path. Note that `main([])` would actually exit via argparse (the positional `name` is required) rather than reaching `_validate_name`, so to genuinely test empty input you'd pass an explicit empty token (e.g. a quoted `""` argv element) and assert SystemExit/exit code. As written this row is dead test weight — either drop `""` from the params or make it call `main` with an argv that reaches the validator.

**Review by mvillmow — COMMENTED** [review]
All 4 issue items correctly addressed; audit task non-recursive, scaffolder well-tested, version-flag comment resolved. One minor tautological test case.

File: hephaestus/scripts_lib/scaffold_subpackage.py [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+    # Refuse to overwrite an existing target
+    pkg_dir = root / "hephaestus" / args.name
+    if pkg_dir.exists():
+        print(
+            f"Error: target directory {pkg_dir} already exists; refusing to overwrite.",
+            file=sys.stderr,
+        )
+        return 1
+
+    if args.dry_run:
```
Comment (mvillmow) [inline hephaestus/scripts_lib/scaffold_subpackage.py:175]: The overwrite guard only checks `hephaestus/<name>`. If that package dir is absent but `tests/unit/<name>/` or `scripts/<name>.py` already exist (e.g. a half-scaffolded or hand-written test dir), the write loop will silently overwrite them. Consider checking every planned target path for existence before writing, not just the package dir.

**Review by mvillmow — COMMENTED** [review]
All 4 issue items addressed correctly (MIGRATION link, _required.yml scope note, non-recursive audit task, scaffolder w/ tests). One minor: overwrite guard only checks package dir, not test/script targets.

**Review by mvillmow — COMMENTED** [review]
LGTM — all 4 issue items addressed; overwrite guard checks all planned paths, prior test-quality nit fixed, doc counts consistent.
